Setup linters

loicteixeira opened the pull request:
Add Linters dependencies & config for Python and JS. Waiting on https://github.com/prettier/prettier/pull/3038 for SASS.

I'll add the steps to the CI after merging.

loicteixeira commented on Makefile:
```diff
@@ -0,0 +1,14 @@
+	find . -name '*~' -exec rm -f {} +
+
+lint-py: ## Run Python linters
+	flake8 $(files)
```
Unfortunately files passed explicitly in the CLI have precedence over the exclude list of the config.
For example, if you create a Django migration, it will be parsed by flake8 (and fail because of line length) despite being ignore.

I don't think we can use `flake8` [default githook](https://gitlab.com/pycqa/flake8/blob/3530870679ddb93ade9326d9e5d932596b66e8b9/src/flake8/main/git.py) but maybe the hook could use Python instead and take advantage of [update_excludes](https://gitlab.com/pycqa/flake8/blob/3530870679ddb93ade9326d9e5d932596b66e8b9/src/flake8/main/git.py#L214-219).

loicteixeira commented on .githooks/pre-commit:
```diff
@@ -0,0 +1,32 @@
+
+for HOOK in $HOOKS
+do
+    source $HOOK "$CHANGED_FILES"
```
In order to allow other scripts than `bash` scripts (e.g. `python` script), we might need to replace this with `$HOOK "$CHANGED_FILES"` and add execution permission to the `pre-commit-*` files.

thibaudcolas replied:
I think it would be reasonable to assume all files in `hooks` are executable. Is this the only reason why you used `source`, or does it do something else?

thibaudcolas commented on .githooks/pre-commit-js:
```diff
@@ -0,0 +1,5 @@
+JS_FILES=$(echo "$CHANGED_FILES" | { grep "\.js$" || true; })
+
+[[ -z "$JS_FILES" ]] && return
```
Ah that's so much cleaner than all the `ifs`!

loicteixeira commented:
As discussed offline, we will look at either using Prettier with the default indenting style (without the PR mentioned above merged) or something else as it is unclear when/if the PR will be merged.

loicteixeira commented:
@thibaudcolas:
- It seems that VisualStudio Code does not respect `.editorconfig` for `Makefile`s and use tabs of size 4 anyway (which is why it worked so far) but I've added the config as not everybody might use VSCode.
- I've made all the hooks executable and removed the `source` but FWIW, we can't use `return` anymore but have to use `exit 0` instead as you can only return from within a function or a sourced script.

loicteixeira commented:
Merging without CSS linting as I can't quite get it to work but need to get on something else. I'll create a separate issue to revisit.

thibaudcolas commented:
🎉 

For me VS Code respects the settings defined for https://github.com/springload/draftjs_exporter, I might have a plugin doing that though not sure.
